pkg/registry/core/pod/storage/eviction.go

```diff
@@ -203,6 +235,27 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 	return &metav1.Status{Status: metav1.StatusSuccess}, nil
 }
 
+// canIgnorePDB returns true for pod conditions that allow the pod to be deleted
+// without checking PDBs.
+func canIgnorePDB(pod *api.Pod) bool {
+	if pod.Status.Phase == api.PodSucceeded || pod.Status.Phase == api.PodFailed || pod.Status.Phase == api.PodPending {
+		return true
+	}
+	return false
+}
+
+func shouldEnforceResourceVersion(pod *api.Pod) bool {
+	// Only pods that may be included as health in PDBs in the future need to be checked.
+	if pod.Status.Phase == api.PodPending {
```

Comment:
is this only run on pods that pass `canIgnorePDB`? would it be clearer to default to enforcing resourceVersion except in terminal cases, e.g.:

```go
if pod.Status.Phase == api.PodSucceeded || pod.Status.Phase == api.PodFailed {
  return false
}
return true
```

Reply:
Yes, this is only run on pods that pass canIgnorePDB.  I think what you're saying makes sense.  You need to specifically opt-out of enforcing ResourceVersion at this point, and default to enforcing otherwise.

I think that's pretty minor, I could do a follow up if you don't think it's a blocker.

Reply:
> Yes, this is only run on pods that pass canIgnorePDB. I think what you're saying makes sense. You need to specifically opt-out of enforcing ResourceVersion at this point, and default to enforcing otherwise.
> 
> I think that's pretty minor, I could do a follow up if you don't think it's a blocker.

it's an easy, minor comment.  Let's just get it cleaned up.